src/Vector.cpp

```diff
@@ -0,0 +1,74 @@
+#include "Vector.h"
+#include <algorithm>
+
+Vector::Vector(size_t initial_size, int initial_value) : m_size(initial_size), m_capacity(initial_size) {
+    m_data = new int[m_capacity];
+    for (size_t i = 0; i < m_size; ++i) {
+        m_data[i] = initial_value;
+    }
+}
+
+Vector::~Vector() {
+    delete[] m_data;
+}
+
+Vector::Vector(const Vector& other) : m_size(other.m_size), m_capacity(other.m_capacity) {
+    m_data = new int[m_capacity];
+    std::copy(other.m_data, other.m_data + m_size, m_data);
+}
+
+Vector& Vector::operator=(const Vector& other) {
+    if (this != &other) {
+        delete[] m_data;
+        m_size = other.m_size;
+        m_capacity = other.m_capacity;
+        m_data = new int[m_capacity];
+        std::copy(other.m_data, other.m_data + m_size, m_data);
+    }
+    return *this;
+}
+
+void Vector::push_back(int value) {
+    if (m_size == m_capacity) {
+        reserve(m_capacity == 0 ? 1 : 2 * m_capacity);
```

Comment:
This `if` can be spared if `m_capacity` is set to larger than 0 from the beginning